Kudu.Core/Infrastructure/ProcessExtensions.cs

```diff
@@ -244,7 +244,14 @@ public static string GetCommandLine(this Process process)
 
         public static bool GetIsScmSite(Dictionary<string, string> environment)
         {
-            return environment[WellKnownEnvironmentVariables.ApplicationPoolId].StartsWith("~1", StringComparison.OrdinalIgnoreCase);
+            string appPool = null;
+            if (environment.TryGetValue(WellKnownEnvironmentVariables.ApplicationPoolId, out appPool) &&
+                !string.IsNullOrEmpty(appPool))
```

Comment:
I'm doing that IsNullOrEmpty check because below we're doing a StartsWith - we don't want to null ref.